channeldb/graph.go

```diff
@@ -605,17 +620,27 @@ func (c *ChannelGraph) addChannelEdge(tx *bbolt.Tx, edge *ChannelEdgeInfo) error
 // was updated for both directed edges are returned along with the boolean. If
 // it is not found, then the zombie index is checked and its result is returned
 // as the second boolean.
-func (c *ChannelGraph) HasChannelEdge(chanID uint64,
-) (time.Time, time.Time, bool, bool, error) {
+func (c *ChannelGraph) HasChannelEdge(
+	chanID uint64) (time.Time, time.Time, bool, bool, error) {
+
+	c.cacheMu.Lock()
```

Comment:
`RLock` for the exist-check?

Reply:
I'd be okay with doing so. We'd have to check twice, once w/ shared lock and then again after acquiring the exclusive, but likely better performance